### ISSUE TITLE
Cluster broken after 1.20.2 update

### ISSUE THREAD

**Issue by oleo65** [ubuntu/microk8s#2014]
After auto upgrading of my nodes to v1.20.2 today they started to become `Not Ready`. I suspected the minor versions to be incompatible so I upgraded all nodes and the master to the most recent snap version.

The master somehow recovered and is displayed as Ready by `kubectl get nodes`. The nodes however did not. They actually did not even send their current version to the master which stayed pre-update.

```
ubuntu@pi-k8s-1:/var/snap/microk8s/current$ kubectl get nodes
NAME       STATUS     ROLES    AGE    VERSION
pi-k8s-1   Ready      <none>   226d   v1.20.2-35+e99a4bd69e2a3e
pi-k8s-2   NotReady   <none>   226d   v1.20.1-34+97978f80232b01
```

I tried to rejoin another node (pi-k8s-3) to the cluster and even did a hard reset on microk8s on that node. But the join process does not succeed even after a fresh install of the snap on node 3. `Failed to join cluster. Error code 500.` There seems something to be really off after the upgrade with the communication and I suspect the errors to be related.

Maybe it is related to the introduction of `kubelite` which were not present before.

Is there some way to recover my cluster maybe by temporarily reverting from `kubelite` to the way it was before?

I am happy to provide more details if needed.

Inspect on master node
```
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-kubelite is running
  Service snap.microk8s.daemon-flanneld is running
  Service snap.microk8s.daemon-etcd is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster
Inspecting juju
  Inspect Juju
Inspecting kubeflow
  Inspect Kubeflow

 WARNING:  Port 16443 seems to be in use by another application.
# Warning: iptables-legacy tables present, use iptables-legacy to see them
Building the report tarball
  Report tarball is at /var/snap/microk8s/2048/inspection-report-20210216_151718.tar.gz
```
[inspection-report-20210216_151718.tar.gz](https://github.com/ubuntu/microk8s/files/5989300/inspection-report-20210216_151718.tar.gz)

Port 16443 is used by kubelite according to `ss -tulwnp` which seems somehow related.

**Comment by ktsakalozos** [comment]
Hi @oleo65 thank you for reporting this.

Indeed, kubelite is the problem here. We have reverted the offending revision so now the nodes should be running with the old setup (no kubelite).

The issue at hand appeared in the context of a non-HA (etcd based) cluster as the one you have here. kubelite would try to start all services, including the control plane but this is not what should happen in the non-HA setup. We are working on this case now.

Something not related to this issue. Your cluster is a long lasting one (209) days. For such use cases you may want to follow a channel other than the latest/stable so you do not get surprising upgrades (eg from 1.20 to 1.21).

**Comment by oleo65** [comment]
Perfect @ktsakalozos thank you alot for your feedback and fast response. Really appreciated.

The cluster started automagically working again after the latest snaps arrived on the nodes. I really like the snap concept for being no frills most of the time.

Thank you also for your feedback related to the specific release channels. I was getting more familiar with snap and kube over the last month and should probably pin it to a specific channel and then intentionally trigger the update to the next major revision.

As far as the current issue is concerned, pinning to 1.20/stable would not have prevented this, right?

**Comment by ktsakalozos** [comment]
> As far as the current issue is concerned, pinning to 1.20/stable would not have prevented this, right?

No it wouldn't. We thought kubelite would have been a drop-in replacement to k8s services but we missed some usecases. It will be released on v1.21.

**Comment by balchua** [comment]
Will be closing this issue.  Since it is already addressed. 
Thanks
